src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/PlaceRecognizer.scala

```diff
@@ -0,0 +1,52 @@
+package com.microsoft.partnercatalyst.fortis.spark.transforms.locations
+
+import java.io.IOError
+
+import com.microsoft.partnercatalyst.fortis.spark.logging.Loggable
+import com.microsoft.partnercatalyst.fortis.spark.transforms.ZipModelsProvider
+import com.microsoft.partnercatalyst.fortis.spark.transforms.nlp.OpeNER
+import ixa.kaflib.Entity
+
+import scala.collection.JavaConversions._
+
+@SerialVersionUID(100L)
+class PlaceRecognizer(
+  modelsSource: Option[String] = None,
+  enabledLanguages: Set[String] = Set("de", "en", "es", "eu", "it", "nl")
+) extends Serializable with Loggable {
+
+  @volatile private lazy val modelsProvider = createModelsProvider()
+
+  def extractPlaces(text: String, language: String): Iterable[String] = {
+    if (!enabledLanguages.contains(language)) {
+      return Set()
+    }
+
+    try {
+      val resourcesDirectory = modelsProvider.ensureModelsAreDownloaded(language)
+
+      val kaf = OpeNER.tokAnnotate(resourcesDirectory, text, language)
+      OpeNER.posAnnotate(resourcesDirectory, language, kaf)
+      OpeNER.nerAnnotate(resourcesDirectory, language, kaf)
+
+      logDebug(s"Analyzed text $text in language $language: $kaf")
+
+      kaf.getEntities.toList.filter(entityIsPlace).map(_.getStr).toSet
+    } catch {
+      case ex @ (_ : NullPointerException | _ : IOError) =>
```

Comment:
I wish it weren't... but unfortunately the OpeNER code is research-quality code so it throws a NPE in some cases when it can't load a model.